### PR TITLE
refactor: redesign functionality for creating KMS instance

### LINKED ISSUE #29: Pass KMS and Keystore services as dependencies to Operation

### LINKED ISSUE #60: Redesign functionality for creating local KMS instance
- [ ] Update SecretLock service for the KMS instance
    - [ ] Should not require passphrase from the user
    - [ ] Reconstruct a full key and use it as a master key for the KMS instance
- [ ] Move `KMSCreator` logic into its own package

### PR DESCRIPTION
**Pull request by aholovko**
The main issue that this PR is aimed to solve is moving the **complexity of creating a KMS instance** out of the composition root (`start.go`). That logic is becoming quite complicated (especially with upcoming support for the key shares) and so requires appropriate test coverage (that is hard to do when it's a part of start.go). The functionality is now encapsulated in the `kms` package within `kms.NewLocalKMS` and `kms.NewKMSServiceCreator` routines (KMS instances are created on per-user basis thus we need to delay creation until the request is made).

Another problem was that request models in `operation` package (`createKeystoreReq`, `createKeyReq`, and so on) were messed up with different concerns:
* payload for the operation itself, e.g. `message` for the signing request and
* lock parameter (`passphrase`) for a secret lock service.

They are two different tasks and better to keep them separate. Now `passphrase` is handled by the layer responsible for creating a KMS service instance for the user (`kms.NewKMSServiceCreator`). And as a result, Operation REST API logic is much cleaner now and deals only with parsing request payloads and preparing responses.

Guys, excuse me for the size of this PR - I was not able to get the full picture until done it as a whole. I will try to be more granular the next time.

Closes #60
Closes #29

Signed-off-by: Andriy Holovko <andriy.holovko@gmail.com>

### CODE REVIEW

File: pkg/internal/common/httphandler_test.go [hunk 1]
```diff
@@ -4,14 +4,16 @@ Copyright SecureKey Technologies Inc. All Rights Reserved.
 SPDX-License-Identifier: Apache-2.0
 */
 
-package support
```
Comment (llorllale) [inline pkg/internal/common/httphandler_test.go:7]: It is unusual having a package not named after the directory it's in.

**Review by llorllale — APPROVED** [review]
LGTM

File: pkg/internal/common/httphandler_test.go (hunk 1 above)
Comment (aholovko) [reply pkg/internal/common/httphandler_test.go:7]: updated

File: pkg/kms/creator.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		p := struct {
+			Passphrase string `json:"passphrase"`
+		}{}
+
+		err = json.NewDecoder(b).Decode(&p)
+		if err != nil {
+			return nil, err
+		}
+
+		secLock, err := hkdf.NewMasterLock(p.Passphrase, sha256.New, nil)
```
Comment (Baha-sk) [inline pkg/kms/creator.go:72]: I guess `p.Passphrase` will be dealt with using secret splitting in a later pr?

**Review by Baha-sk — APPROVED** [review]

File: pkg/kms/creator.go (hunk 2 above)
Comment (aholovko) [reply pkg/kms/creator.go:72]: yep

**Pull request merged**